src/routes/safe/components/Transactions/helpers/TxParamsState/index.tsx

```diff
@@ -0,0 +1,163 @@
+import { ReactNode, useState } from 'react'
+import styled from 'styled-components'
+import { EstimationStatus, useEstimateTransactionGas } from 'src/logic/hooks/useEstimateTransactionGas'
+import { TxParameters } from 'src/routes/safe/container/hooks/useTransactionParameters'
+import { EditableTxParameters } from 'src/routes/safe/components/Transactions/helpers/EditableTxParameters'
+import { extractSafeAddress } from 'src/routes/routes'
+import { ReviewInfoText } from 'src/components/ReviewInfoText'
+import ExecuteCheckbox from 'src/components/ExecuteCheckbox'
+import { useEstimationStatus } from 'src/logic/hooks/useEstimationStatus'
+import { Errors, logError } from 'src/logic/exceptions/CodedException'
+import { ButtonStatus, Modal } from 'src/components/Modal'
+import { lg, md } from 'src/theme/variables'
+import { TxParametersDetail } from '../TxParametersDetail'
+
+type Props = {
+  children: ReactNode
+  txData: string
+  txValue?: string
+  txType?: string
+  onSubmit: (txParams: TxParameters, delayExecution?: boolean) => void
+  onBack?: (...rest: any) => void
+  submitText?: string
+}
+
+const Container = styled.div`
+  padding: 0 ${lg} ${md};
+`
+
+export const TxParamsState = ({
+  children,
+  txData,
+  txValue = '0',
+  txType,
+  onSubmit,
+  onBack,
+  submitText,
+}: Props): React.ReactElement => {
+  const [manualSafeTxGas, setManualSafeTxGas] = useState('0')
+  const [manualGasPrice, setManualGasPrice] = useState<string | undefined>()
+  const [manualGasLimit, setManualGasLimit] = useState<string | undefined>()
+  const [executionApproved, setExecutionApproved] = useState<boolean>(true)
+  const safeAddress = extractSafeAddress()
+  const isSpendingLimit = txType === 'spendingLimit'
```

Comment:
Here is an example where my aforementioned helper function could be used instead.